src/node_modules/app/machines/builder/form.options.js

```diff
@@ -159,6 +213,38 @@ function computeDefaultValues (ctx) {
 	}
 }
 
+function processRequestValue (params, name, value, type) {
+	if (!value) {
+		return
+	} else if (is_object(type) || is_xor(type) || is_extent(type)) {
+		if (isObjEmpty(value)
+			&& !(
+				is_optional(type)
+				&& '__default' in type
+				&& isObjEmpty(type.__default)
+			)
+		) {
+			delete params[name];
+		}
+		const info = getTypings(type);
+		info.forEach(prop => {
+			processRequestValue(
+				value,
+				prop.name,
+				value[prop.name],
+				prop.type
+			);
+		});
+	} else if (is_intWithUnit(type)) {
+		params[name] = `${value.__value}${value.__unit}`;
+	}
+	if (is_xor(type)) {
+		const fieldName = value.__selection;
+		params[fieldName] = params[name][fieldName];
+		delete params[name];
+	}
+}
```

Comment:
This whole function has to be rewritten as it modifies the arguments. Functions should return values instead.

Reply:
I agree, working on it.